File: apps/mobile/lib/features/logbook/domain/entities/trip.dart
class Trip {
  const Trip({
    required this.id,
    required this.boatId,
    required this.departurePort,
    required this.departureTime,
    this.arrivalPort,
    this.arrivalTime,
    this.distanceNm,
    this.maxSpeedKnots,
    this.avgSpeedKnots,
    this.notes,
    this.trackPoints,
    this.status = TripStatus.completed,
    this.engineHours,
    this.fuelConsumedL,
    this.crewMembers,
    this.createdAt,
    this.updatedAt,
  });

  final String id;
  final String boatId;
  final String departurePort;
  final DateTime departureTime;
  final String? arrivalPort;
  final DateTime? arrivalTime;
  final double? distanceNm;
  final double? maxSpeedKnots;
  final double? avgSpeedKnots;
  final String? notes;
  final List<TrackPoint>? trackPoints;
  final TripStatus status;
  final double? engineHours;
  final double? fuelConsumedL;
  final List<String>? crewMembers;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Duration? get duration {
    if (arrivalTime == null) return null;
    return arrivalTime!.difference(departureTime);
  }

  Trip copyWith({
    String? id,
    String? boatId,
    String? departurePort,
    DateTime? departureTime,
    String? arrivalPort,
    DateTime? arrivalTime,
    double? distanceNm,
    double? maxSpeedKnots,
    double? avgSpeedKnots,
    String? notes,
    List<TrackPoint>? trackPoints,
    TripStatus? status,
    double? engineHours,
    double? fuelConsumedL,
    List<String>? crewMembers,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Trip(
      id: id ?? this.id,
      boatId: boatId ?? this.boatId,
      departurePort: departurePort ?? this.departurePort,
      departureTime: departureTime ?? this.departureTime,
      arrivalPort: arrivalPort ?? this.arrivalPort,
      arrivalTime: arrivalTime ?? this.arrivalTime,
      distanceNm: distanceNm ?? this.distanceNm,
      maxSpeedKnots: maxSpeedKnots ?? this.maxSpeedKnots,
      avgSpeedKnots: avgSpeedKnots ?? this.avgSpeedKnots,
      notes: notes ?? this.notes,
      trackPoints: trackPoints ?? this.trackPoints,
      status: status ?? this.status,
      engineHours: engineHours ?? this.engineHours,
      fuelConsumedL: fuelConsumedL ?? this.fuelConsumedL,
      crewMembers: crewMembers ?? this.crewMembers,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Trip && runtimeType == other.runtimeType && id == other.id;

  @override
  int get hashCode => id.hashCode;
}

enum TripStatus { recording, paused, completed }

class TrackPoint {
  const TrackPoint({
    required this.latitude,
    required this.longitude,
    required this.timestamp,
    this.speedKnots,
  });

  final double latitude;
  final double longitude;
  final DateTime timestamp;
  final double? speedKnots;
}

File: apps/api/internal/domain/trip.go
package domain

import "time"

// TripStatus represents the current status of a trip.
type TripStatus string

// TripStatus values.
const (
	TripStatusPlanned   TripStatus = "planned"
	TripStatusRecording TripStatus = "recording"
	TripStatusCompleted TripStatus = "completed"
	TripStatusCancelled TripStatus = "cancelled"
)

// TripKind distinguishes a regular trip from a group regatta.
type TripKind string

// TripKind values.
const (
	TripKindTrip    TripKind = "trip"
	TripKindRegatta TripKind = "regatta"
)

// RSVP represents a participant's attendance answer to a planned group trip.
type RSVP string

// RSVP values.
const (
	RSVPGoing    RSVP = "going"
	RSVPMaybe    RSVP = "maybe"
	RSVPNotGoing RSVP = "not_going"
)

// Trip represents a sailing trip, passage, or group regatta.
type Trip struct {
	ID                   string
	BoatID               string
	UserID               string
	GroupID              *string
	Title                *string
	Kind                 TripKind
	ScheduledAt          *time.Time
	ChecklistCompletedAt *time.Time
	DeparturePort        string
	ArrivalPort          *string
	DepartureTime        time.Time
	ArrivalTime          *time.Time
	DistanceNM           *float64
	MaxSpeedKnots        *float64
	AvgSpeedKnots        *float64
	EngineHours          *float64
	FuelConsumedL        *float64
	DurationMinutes      *int
	CrewMembers          []string
	WeatherConditions    map[string]any
	Notes                *string
	Photos               []string
	Status               TripStatus
	CreatedAt            time.Time
	UpdatedAt            time.Time
}

// TripParticipant represents a user's RSVP to a planned group trip/regatta.
type TripParticipant struct {
	TripID      string
	UserID      string
	RSVP        RSVP
	RespondedAt time.Time
}

// ChecklistItem represents a single safety-checklist entry attached to a trip.
type ChecklistItem struct {
	ID        string
	TripID    string
	Label     string
	IsChecked bool
	Position  int
	CreatedAt time.Time
	UpdatedAt time.Time
}

// TripTrack represents a single GPS track point recorded during a trip.
type TripTrack struct {
	ID         string
	TripID     string
	Lat        float64
	Lon        float64
	SpeedKnots *float64
	Heading    *float64
	RecordedAt time.Time
}
